src/Maestro/tests/Maestro.ScenarioTests/MaestroScenarioTestBase.cs

```diff
@@ -323,5 +449,87 @@ public async Task<string> GetRepositoryPolicies(string repoUri, string branchNam
         {
             return await RunDarcAsync("get-repository-policies", "--all", "--repo", repoUri, "--branch", branchName);
         }
+
+        public void CheckGitDirectory(string path, string gitDirFolder)
+        {
+            string[] expectedFolders = { "hooks", "info", "logs", "objects", "refs" };
+            foreach (string folder in expectedFolders)
+            {
+                string folderPath = Path.Combine(gitDirFolder, folder);
+                Assert.IsTrue(Directory.Exists(folderPath), $"{path} does not appear to be a valid .gitdir: missing folder '{folder}'");
+            }
+
+            string[] expectedFiles = { "config", "description", "FETCH_HEAD", "HEAD", "index" };
+            foreach (string file in expectedFiles)
+            {
+                string filePath = Path.Combine(gitDirFolder, file);
+                Assert.IsTrue(File.Exists(filePath), $"{path} does not appear to be a valid .gitdir: missing file '{file}'");
+            }
+        }
+
+        public void VerifyClonedDirectories(Dictionary<string, string> repos, List<string> masterRepos, List<string> gitDirs, string gitDirFolder, string reposFolder)
+        {
+            foreach (string name in repos.Keys)
+            {
+                string path = Path.Combine(reposFolder, name);
+                bool checkDir = Directory.Exists(path);
+                Assert.IsTrue(Directory.Exists(path), $"Expected cloned repo '{name}' but not found at '{path}'");
+
+                string versionDetailsPath = Path.Combine(path, "eng", "Version.Details.xml");
+                if (File.Exists(versionDetailsPath))
+                {
+                    using (FileStream stream = File.OpenRead(versionDetailsPath))
```

Comment:
NUnit has a "FileAssert.AreEqual", so we could just check the file to be the right value.